admin/add_copyright.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env PYTHONPATH=./gcovr python3
+
+# -*- coding:utf-8 -*-
+
+#  ************************** Copyrights and license ***************************
+#
+# This file is part of gcovr 4.2, a parsing and reporting tool for gcov.
+# https://gcovr.com/en/stable
+#
+# _____________________________________________________________________________
+#
+# Copyright (c) 2021 the gcovr authors
+# Copyright (c) 2013 Sandia Corporation.
+# This software is distributed under the BSD License.
+# Under the terms of Contract DE-AC04-94AL85000 with Sandia Corporation,
+# the U.S. Government retains certain rights in this software.
+# For more information, see the README.rst file.
+#
+# ****************************************************************************
+
+import os
+import re
+import subprocess
+
+import version
+
+REGEX_EMAIL = re.compile(r"\d+\+([^@]+)@users\.noreply\.github\.com")
+HEADER_END = (
+    " ****************************************************************************"
+)
+
+
+def getLicenseSection(filename, comment_char="#"):
+    date = subprocess.check_output(
+        ["git", "log", "-1", "--format=format:%ad", "--date=short", "--", filename],
+        universal_newlines=True,
+    )
+    year = date[:4]
+    yield comment_char + "  ************************** Copyrights and license ***************************"
+    yield comment_char
+    yield comment_char + " This file is part of gcovr {}, a parsing and reporting tool for gcov.".format(
+        version.__version__
+    )
+    yield comment_char + " https://gcovr.com/en/stable"
+    yield comment_char
+    yield comment_char + " _____________________________________________________________________________"
+    yield comment_char
+    yield comment_char + " Copyright (c) " + year + " the gcovr authors"
```

Comment:
I'd probably keep a year range to indicate that there were changes in between 2013 and now. Most files include some of that old code, though this is not quite correct for the newer generators. On the other hand, `git log --follow` isn't particularly accurate either.

```suggestion
    yield comment_char + f" Copyright (c) 2013-{year} the gcovr authors"
```

Reply:
I'll change this to `"".format()` and after the next release to `f"..."`.

Reply:
Cahanged to 2013-<Year of last change>.